src/Merge/MultiConstraint.php

```diff
@@ -65,10 +68,13 @@ private static function optimizeConstraints(array $constraints, $conjunctive)
             $optimized = false;
             for ($i = 1, $l = \count($constraints); $i < $l; $i++) {
                 $right = $constraints[$i];
-                if ($left instanceof MultiConstraint
+                if (
+                    $left instanceof self
                     && $left->conjunctive
-                    && $right instanceof MultiConstraint
+                    && $right instanceof self
```

Comment:
I guess this should be `SemverMultiConstraint`

Reply:
Thank you!